platformics/codegen/tests/test_file_mutations.py

```diff
@@ -214,3 +215,93 @@ async def test_create_file(
     """
     output = await gql_client.query(mutation, member_projects=[123], service_identity="workflows")
     assert output["data"]["createFile"]["size"] == file_size
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "file_path,multiple_files_for_one_path,should_delete",
+    [
+        ("nextgen/test1.fastq", False, True),
+        ("bla/test1.fastq", False, False),
+        ("nextgen/test1.fastq", True, False),
+        ("bla/test1.fastq", True, False),
+    ],
+)
+async def test_delete_from_s3(
+    file_path: str,
+    should_delete: bool,
+    multiple_files_for_one_path: bool,
+    sync_db: SyncDB,
+    gql_client: GQLTestClient,
+    moto_client: S3Client,
+    monkeypatch: typing.Any,
+) -> None:
+    """
+    Test that we delete a file from S3 under the right circumstances
+    """
+    user1_id = 12345
+    project1_id = 123
+    user2_id = 67890
+    project2_id = 456
+    bucket = "local-bucket"
```

Comment:
the bucket name should probably be a pytest fixture as well when we get a chance to switch it -- each test could create/use its own temporary bucket so parallel tests won't get in each others' way